### PR TITLE
Renesas patch

### PR DESCRIPTION
**Pull request by budaeva**
build fix, RCPML returns composite, launch with newer eclipse, combo databinding fix, onchange attr and listeners for scripting, enabled attr added

### CODE REVIEW

**Comment by a-bastrykina** [comment]
Please remove .DS_Store and  org.rcmpl.example.rss/.DS_Store

**Comment by a-bastrykina** [comment]
Do we need org.apache.batik.dom plugin with org.apache.batik.dom/lib/batik-dom-1.11.jar? I assume both batik 1.11 and 1.9 are taken from the target platform

File: org.rcpml.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -22,3 +27,5 @@ Export-Package: org.rcpml.core,
  org.rcpml.core.internal.contentprovider;x-friends:="org.rcpml.core.tests",
  org.rcpml.core.utils,
  org.rcpml.core.xml
+Import-Package: org.eclipse.swt.widgets,
```
Comment (a-bastrykina) [inline org.rcpml.core/META-INF/MANIFEST.MF]: Use bundle import instead of package import

File: org.rcpml.core/src/org/rcpml/core/RCPML.java [hunk 2]
```diff
@@ -33,4 +36,27 @@ public static IController create(Document document) {
...
 		return new Controller(document, true);
 	}
+	
+	public static Composite createComposite(Document doc, Composite parent) {
+		Object presentation = RCPML.renderDocument(doc);
+		Composite composite = getComposite(presentation, parent);
+		return composite;
+	}
+	
+	private static Composite getComposite(Object presentation, Composite parent) {
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/RCPML.java:47]: This getter method has a side-effect (changing parent composite)
Move parent  composite.setParent(parent); into caller method.

File: org.rcpml.core.tests/.classpath [hunk 3]
```diff
@@ -1,7 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER">
+		<attributes>
+			<attribute name="module" value="true"/>
+		</attributes>
+	</classpathentry>
```
Comment (a-bastrykina) [inline org.rcpml.core.tests/.classpath]: Seems like this change is not needed

File: org.rcpml.core/src/org/rcpml/core/RCPML.java [hunk 4]
```diff
@@ -33,4 +36,27 @@ public static IController create(Document document) {
 	public static IController createWithConstructor(Document document) {
 		return new Controller(document, true);
 	}
+	
+	public static Composite createComposite(Document doc, Composite parent) {
+		Object presentation = RCPML.renderDocument(doc);
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/RCPML.java:41]: [minor] Get rid of Object usage (use common interfaces)

File: org.rcpml.core/src/org/rcpml/core/internal/Controller.java [hunk 5]
```diff
@@ -27,6 +27,7 @@
 import org.w3c.dom.events.EventTarget;
 
 import com.xored.scripting.core.IScriptContextManager;
+//import com.xored.scripting.core.ScriptContextManager;
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/Controller.java]: Delete comment

File: org.rcpml.core/src/org/rcpml/core/internal/Controller.java [hunk 6]
```diff
@@ -73,6 +74,7 @@ public Controller(Document document, boolean withConstructor) {
 
 		this.fScriptContextManager = ScriptingCore.createContextManager();
 		this.fScriptContextManager.addDefaultBinding("document", this.fDocument);
+//		this.fScriptContextManager = new ScriptContextManager(this.fDocument);
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/Controller.java]: Delete commented code

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMDocument.java [hunk 7]
```diff
@@ -268,5 +270,161 @@ public void putStyleMapForElement( Element element, StyleMap styleMap ) {
 	public void clearStylesMap() {
 		this.fStyleMapHash.clear();
 	}
+
+	@Override
+	public String getInputEncoding() {
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMDocument.java]: Where is this class used? Won't the empty implementation cause bugs?

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMElement.java [hunk 8]
```diff
@@ -42,21 +46,26 @@ public String getCSSClass() {
...
-			// !!! TODO
-			e.printStackTrace();
-			throw new InternalError();
-		}
+			return new ParsedURL(bu);
+//		} catch (MalformedURLException e) {
+//			// !!! TODO
+//			e.printStackTrace();
+//			throw new InternalError();
+//		}
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMElement.java]: Please remove commented code

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMElement.java [hunk 9]
```diff
@@ -42,21 +46,26 @@ public String getCSSClass() {
 		return getAttributeNS(null, "class");
 	}
 
-	public URL getCSSBase() {
-		try {
+	public ParsedURL getCSSBase() {
+//		try {
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMElement.java]: Please remove commented code

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMElement.java [hunk 10]
```diff
@@ -42,21 +46,26 @@ public String getCSSClass() {
...
-		}
+			return new ParsedURL(bu);
+//		} catch (MalformedURLException e) {
+//			// !!! TODO
+//			e.printStackTrace();
+//			throw new InternalError();
+//		}
+	}
+
+	public Element getXblBoundElement() {
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMElement.java]: Please check unimplemented method usage

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMStyleElement.java [hunk 11]
```diff
@@ -78,17 +78,17 @@ public StyleSheet getCSSStyleSheet() {
...
-	                } catch (MalformedURLException ex) {
-	                    // !!! TODO
-	                    ex.printStackTrace();
-	                    throw new InternalError();
-	                }
+//	                } catch (MalformedURLException ex) {
+//	                    // !!! TODO
+//	                    ex.printStackTrace();
+//	                    throw new InternalError();
+//	                }
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMStyleElement.java]: Delete commented code

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMStyleElement.java [hunk 12]
```diff
@@ -78,17 +78,17 @@ public StyleSheet getCSSStyleSheet() {
 	                    }
 	                    text = sb.toString();
 	                }
-	                URL burl = null;
-	                try {
+	                ParsedURL burl = null;
+//	                try {
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMStyleElement.java]: Delete commented code

File: org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMStyleElement.java [hunk 13]
```diff
@@ -103,6 +103,13 @@ public StyleSheet getCSSStyleSheet() {
 	}
 
 
+	public void addEventListenerNS(String xmlEventsNamespaceUri, String string,
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/dom/RCPOMStyleElement.java]: Please check unimplemented method usage

File: org.rcpml.core/src/org/rcpml/core/internal/scripting/ScriptBridge.java [hunk 14]
```diff
@@ -14,6 +14,7 @@
 import org.rcpml.core.internal.contentprovider.ContentProviderManager;
 import org.w3c.dom.Element;
 import org.w3c.dom.Node;
+import org.w3c.dom.events.Event;
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/internal/scripting/ScriptBridge.java]: Organize imports

File: org.rcpml.emf.example/src/org/ecpml/emf/example/JLoaderTagFactory.java [hunk 15]
```diff
@@ -24,12 +24,14 @@ protected JLoaderBridge(Node node, IController container) {
...
-						).bindObject("emfLoader", new EMFBindingManager(container));
-			} catch (ScriptException e) {
-				e.printStackTrace();
-			}
+				try {
+					getController().getScriptManager().getDefaultContext(
+							).bindObject("emfLoader", new EMFBindingManager(container));
+				} catch (ScriptException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
```
Comment (a-bastrykina) [inline org.rcpml.emf.example/src/org/ecpml/emf/example/JLoaderTagFactory.java]: Use activator logging facility:
```
Status status = new Status(IStatus.ERROR, Activator.PLUGIN_ID, "Message", e);
Activator.getDefault().getLog().log(status);
```

File: org.rcpml.forms.tests/.classpath [hunk 16]
```diff
@@ -3,5 +3,6 @@
 	<classpathentry kind="src" path="src"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/3"/>
```
Comment (a-bastrykina) [inline org.rcpml.forms.tests/.classpath]: Is this change needed?

File: org.rcpml.forms/src/org/rcpml/forms/internal/tags/HyperLinkTagFactory.java [hunk 17]
```diff
@@ -41,11 +41,12 @@ public void linkActivated(HyperlinkEvent e) {
...
-						try {
-							getController().getScriptManager().executeScript(script);
-						} catch (ScriptException e1) {
-							e1.printStackTrace();
-						}
+							try {
+								getController().getScriptManager().executeScript(script);
+							} catch (ScriptException e1) {
+								// TODO Auto-generated catch block
+								e1.printStackTrace();
```
Comment (a-bastrykina) [inline org.rcpml.forms/src/org/rcpml/forms/internal/tags/HyperLinkTagFactory.java]: Use Activator logging (see comment above)

File: org.rcpml.swt/META-INF/MANIFEST.MF [hunk 18]
```diff
@@ -7,9 +7,11 @@ Bundle-Vendor: Xored software, Inc
...
 Require-Bundle: org.rcpml.core,
- org.apache.batik,
  org.eclipse.swt,
  org.eclipse.jface.databinding,
  org.eclipse.jface,
- org.eclipse.core.databinding
+ org.eclipse.core.databinding,
+ com.xored.scripting.core;bundle-version="1.0.0",
+ org.apache.batik.css;bundle-version="1.9.1",
+ org.eclipse.core.databinding.property;bundle-version="1.8.100"
```
Comment (a-bastrykina) [inline org.rcpml.swt/META-INF/MANIFEST.MF]: Remove org.eclipse.core.databinding.property version

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 19]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
-		dbc.bindValue(SWTObservables.observeSelection(this.fCombo),
-				new ElementAttributeObservable(getNode(), "state"), null, null);
 
-		update();
 	}
 
+
 	protected Combo constructCombo(Composite parent, int style) {
 		return new Combo(parent, style);
 	}
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java:102]: Add @Override annotation

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 20]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
 		int style = SWT.BORDER | SWT.SINGLE | SWT.DROP_DOWN;
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java:50]: add @Override annotation

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 21]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
 	}
 
+
 	protected Combo constructCombo(Composite parent, int style) {
 		return new Combo(parent, style);
 	}
 
 	public void update() {
 		if (this.fCombo != null) {
+			int oldSelection = this.fCombo.getSelectionIndex();
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: Check usage

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 22]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
 			this.fCombo.setLayoutData(constructLayoutData(fCombo.getParent()));
 //			this.fCombo.setText(DOMUtils.getChildrenAsText(this.getNode()));
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: delete commented code

File: org.rcpml.forms/src/org/rcpml/forms/internal/tags/ComboTagFactory.java [hunk 23]
```diff
@@ -22,6 +22,15 @@ public ComboBridge(Node node, IController container) {
...
 			Combo combo = new Combo(parent, style);
+			String label = getAttribute("label");
+			combo.setToolTipText(label);
+			
+			String items = getAttribute("items");
+			if (items != null) {
+				for (String item : items.split(",")) {
+					combo.add(item);
+				}
+			}
```
Comment (a-bastrykina) [inline org.rcpml.forms/src/org/rcpml/forms/internal/tags/ComboTagFactory.java]: Is this code needed? Combo options are described using child <state> elements

File: org.rcpml.core/src/org/rcpml/core/datasource/DataSourceElementContentBinding.java [hunk 24]
```diff
@@ -16,14 +17,14 @@
  * 
  */
 public class DataSourceElementContentBinding extends AbstractDataSourceElementBinding implements EventListener {
-	private Node fNode;
-	private Object fValue;
+	protected Node fNode;
+	protected Object fValue;
```
Comment (a-bastrykina) [inline org.rcpml.core/src/org/rcpml/core/datasource/DataSourceElementContentBinding.java]: Revert if this not needed

File: org.rcpml.swt/src/org/rcpml/swt/AbstractSWTBridge.java [hunk 25]
```diff
@@ -6,6 +6,7 @@
 import org.rcpml.core.RCPMLException;
 import org.rcpml.core.bridge.AbstractBridge;
 import org.rcpml.core.bridge.IBridge;
+import org.rcpml.core.bridge.ICompositeHolder;
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/AbstractSWTBridge.java:10]: Organize imports

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 26]
```diff
@@ -97,15 +158,25 @@ public void update() {
 			}
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java:147]: [minor] Please refactor this using Set (compare if two sets are equal)

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 27]
```diff
@@ -97,15 +158,25 @@ public void update() {
...
 			if (!gfound) {
 				this.fCombo.setItems(items3);
-				this.fCombo.select(selection);
+			} 
+			if (selection/2 != fCombo.getSelectionIndex() && !state.equals(""))
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: Please remove this. selection/2 will not always work for particular inputs.
To calculate selection, create and remember a counter, which counting Element nodes.

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 28]
```diff
@@ -97,15 +158,25 @@ public void update() {
...
+				this.fCombo.select(selection/2);
+			String enabled = RCPCSSConstants.TRUE_VALUE;
+			enabled = getAttribute(RCPMLTagConstants.ENABLED_ATTR);
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: [minor] Create a method accepting default value: getAttribute(name, defaultValue)

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 29]
```diff
@@ -97,15 +158,25 @@ public void update() {
...
+			if (enabled != null) {
+				if (enabled.equals(RCPCSSConstants.FALSE_VALUE)) {
+					this.fCombo.setEnabled(false);
+				} else {
+					this.fCombo.setEnabled(true);
+				}
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]:
```suggestion
                              this.fCombo.setEnabled(RCPCSSConstants.TRUE_VALUE.equals(enabled)))
```

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 30]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
+			getController().bind(new DataBinding(
+					new DataSourceElementAttributeBinding(getNode(), "state"), path));
+		}
+
+	}
+	
+	protected void initHandlers() {
+		Combo combo = this.fCombo;
+		
+		combo.addModifyListener(new ModifyListener() {
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: Use lambda instead of constructor

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 31]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
+			
+			public void modifyText(ModifyEvent event) {
+				String onChangeAction = getAttribute(RCPMLTagConstants.ONCHANGE_ATTR);
+				if (onChangeAction.length() > 0) {
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: Can onChangeAction be null?

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 32]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
+					IScriptContextManager manager = getController().getScriptManager();
+					IScriptingContext context;
+					try {
+						context = manager.getContextFrom(onChangeAction);
+					} catch (ScriptException e) {
+						e.printStackTrace();
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: Use activator for logging

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 33]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
+						context = null;
+					}
+					if (context != null) {
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java:79]: Should we continue work if exception is thrown? I suggest throwing exception further or move 
```
context.bindObject("node", getNode());
context.executeScript(onChangeAction);
```
inside try block

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java [hunk 34]
```diff
@@ -34,26 +49,72 @@ protected void construct(Composite parent) {
...
 
-		this.fCombo.addDisposeListener(new DisposeListener() {
+		initHandlers();
+		
+		DataBindingContext dbc = this.getBindingContext();
+
+//		dbc.bindValue(SWTObservables.observeSelection(this.fCombo),
+//				new ElementAttributeObservable(getNode(), "state"), null, null);
+		
+		dbc.bindValue(WidgetProperties.comboSelection().observe(fCombo), new ElementAttributeObservable(getNode(), "state"), null, null);
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: [minor] Move "state" to static final constant

File: org.rcpml.swt/src/org/rcpml/swt/SWTSpinnerBridge.java [hunk 35]
```diff
@@ -63,6 +68,38 @@ public void widgetDisposed(DisposeEvent e) {
 		}
 	}
 	
+	protected void initHandlers() {
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTSpinnerBridge.java:72]: Same comments as for SWTComboBridge

File: org.rcpml.swt/src/org/rcpml/swt/SWTSpinnerBridge.java [hunk 36]
```diff
@@ -91,6 +128,16 @@ public void update() {
 			}
 			if (this.fSpinner.getSelection() != value)
 				this.fSpinner.setSelection(value);
+			
+			String enabled = TRUE_VALUE;
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTSpinnerBridge.java]: Same comments as for SWTComboBridge

File: org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java [hunk 37]
```diff
@@ -42,35 +53,67 @@ protected void construct(Composite parent) {
...
 		String path = getAttribute(PATH_ATTR);
 		if (path != null && !path.equals("")) {
-			getController().bind(new DataBinding(
-					new DataSourceElementContentBinding(
-							getNode(), String.class), path));
+			getController().bind(new DataBinding(new DataSourceElementContentBinding(getNode(), String.class), path));
 		}
 	}
 
+	protected void initHandlers() {
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java:65]: Same comments as for SWTComboBridge

File: org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java [hunk 38]
```diff
@@ -42,35 +53,67 @@ protected void construct(Composite parent) {
...
+	}
+
 	protected int getStyle() {
 		int style = 0;
 		RCPStylableElement stylable = (RCPStylableElement) getNode();
-		
+
 		String multiline = getAttribute(MULTILINE);
-		if( multiline != null && multiline.equals(RCPCSSConstants.TRUE_VALUE)) {
+		if (multiline != null && multiline.equals(RCPCSSConstants.TRUE_VALUE)) {
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java]:
```suggestion
		if (RCPCSSConstants.TRUE_VALUE.equals(multiline)) {
```

File: org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java [hunk 39]
```diff
@@ -98,6 +140,16 @@ public void update() {
 					this.fText.setEditable(true);
 				}
 			}
+
+			String enabled = TRUE_VALUE;
```
Comment (a-bastrykina) [inline org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java]: Same comments as for SWTComboBridge

File: org.rcpml.tools.ui/src/org/rcpml/tools/ui/internal/browser/RCPMLBrowser.java [hunk 40]
```diff
@@ -178,8 +177,8 @@ private void renderDocument(Document document) {
 				status.setText("Could not create rcpml");
 				return;
 			}
-			bridgeController.getScriptManager().addDefaultBinding("viewer",
-					support);
+//			bridgeController.getScriptManager().addDefaultBinding("viewer",
```
Comment (a-bastrykina) [inline org.rcpml.tools.ui/src/org/rcpml/tools/ui/internal/browser/RCPMLBrowser.java]: Delete or return commented code

File: org.rcpml.tools.ui/src/org/rcpml/tools/ui/internal/browser/RCPMLBrowser.java [hunk 41]
```diff
@@ -163,8 +163,7 @@ private void renderDocument(Document document) {
 		}
 		IRCPMLConstructor constructor = bridgeController.getRootBridge();
 		if (constructor != null) {
-			bridgeController.getScriptManager().addDefaultBinding("viewer",
-					support);
+//			bridgeController.getScriptManager().addDefaultBinding("viewer", support);
```
Comment (a-bastrykina) [inline org.rcpml.tools.ui/src/org/rcpml/tools/ui/internal/browser/RCPMLBrowser.java]: Delete or return commented code

File: org.rcpml.ui.editors/src/org/rcpml/ui/internal/editors/BridgeEditPart.java [hunk 42]
```diff
@@ -38,7 +38,7 @@ public void createPartControl(Composite parent) {
 		try {
 			this.bridge.getController().getScriptManager().getDefaultContext();
 		} catch (ScriptException e) {
-			// TODO Auto-generated catch block
+//			 TODO Auto-generated catch block
 			e.printStackTrace();
```
Comment (a-bastrykina) [inline org.rcpml.ui.editors/src/org/rcpml/ui/internal/editors/BridgeEditPart.java]: Use activator logging

File: org.rcpml.ui.editors/src/org/rcpml/ui/internal/editors/EditorBridge.java [hunk 43]
```diff
@@ -32,7 +32,7 @@ public void executeInitScript( EditorPart part ) {
 		try {
 			this.getController().getScriptManager().getDefaultContext().bindObject("editor", part );
 		} catch (ScriptException e) {
-			// TODO Auto-generated catch block
+//			 TODO Auto-generated catch block
 			e.printStackTrace();
```
Comment (a-bastrykina) [inline org.rcpml.ui.editors/src/org/rcpml/ui/internal/editors/EditorBridge.java]: Use activator logging

File: org.rcpml.ui/src/org/rcpml/ui/internal/application/ApplicationBridge.java [hunk 44]
```diff
@@ -1,6 +1,8 @@
 package org.rcpml.ui.internal.application;
 
-import org.eclipse.core.runtime.IPlatformRunnable;
+import org.eclipse.equinox.app.IApplication;
+import org.eclipse.equinox.app.IApplicationContext;
+//import org.eclipse.core.runtime.IPlatformRunnable;
```
Comment (a-bastrykina) [inline org.rcpml.ui/src/org/rcpml/ui/internal/application/ApplicationBridge.java]: Organize imports

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java (hunk 19 above)
Comment (budaeva) [reply org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java:102]: added to construct(Composite parent), update(), getPresentation() and dispose() methods

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java (hunk 29 above)
Comment (budaeva) [reply org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: added getEnabled() to AbstractSWTBridge

File: org.rcpml.swt/src/org/rcpml/swt/SWTSpinnerBridge.java (hunk 36 above)
Comment (budaeva) [reply org.rcpml.swt/src/org/rcpml/swt/SWTSpinnerBridge.java]: added getEnabled() to AbstractSWTBridge

File: org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java (hunk 39 above)
Comment (budaeva) [reply org.rcpml.swt/src/org/rcpml/swt/SWTTextBridge.java]: added getEnabled() to AbstractSWTBridge

File: org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java (hunk 31 above)
Comment (budaeva) [reply org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java]: added check

File: org.rcpml.forms/src/org/rcpml/forms/internal/tags/ComboTagFactory.java (hunk 23 above)
Comment (budaeva) [reply org.rcpml.forms/src/org/rcpml/forms/internal/tags/ComboTagFactory.java]: We have already sent examples using "items" attribute instead of "state" children elements to Renesas. Anyway it works only for combos without data binding, so I removed it

File: org.rcpml.core/src/org/rcpml/core/RCPML.java (hunk 4 above)
Comment (budaeva) [reply org.rcpml.core/src/org/rcpml/core/RCPML.java:41]: The only one common class for presentation is Object. In some cases getPresentation() method returns Comrosite objects, in other implementors of ICompositeHolder, but also it can return IDataSource, Application, just null, IPerspectiveFactory, etc.
